dashboard/modules/job/job_manager.py

```diff
@@ -325,6 +326,37 @@ async def _polling(self, child_process: subprocess.Popen) -> int:
                 # still running, yield control, 0.1s by default
                 await asyncio.sleep(self.SUBPROCESS_POLL_PERIOD_S)
 
+    async def _poll_all(self, pid: int):
+        """Poll process and all its child processes until all are completed."""
+        parent_process = psutil.Process(pid)
+        children = parent_process.children(recursive=True)
+
+        while True:
+            (_, alive) = psutil.wait_procs([parent_process] + children, timeout=0)
+            if len(alive) == 0:
+                return
+            else:
+                await asyncio.sleep(self.SUBPROCESS_POLL_PERIOD_S)
+
+    def _kill_job(self, entrypoint_pid: int, sig: signal.Signals):
+        """Kills job associated with pid of process from _exec_entrypoint (Unix only).
+        
+        On Mac OSX systems, this kills the parent process itself along with all of its
+        child processes.
+
+        On Linux systems, the parent process is a shell process used to start the actual
+        entrypoint command, so this kills all its child processes but not the parent
+        process itself.
+        """
```

Comment:
Someone without context might ask "why not kill the parent process in both cases." Can you add the reason to the docstring?

Reply:
Changed approach, killing parent process in both cases now.